lib/dbservice_web/controllers/group_user_controller.ex

```diff
@@ -105,6 +105,35 @@ defmodule DbserviceWeb.GroupUserController do
     end
   end
 
+  def update_by_type(conn, params) do
+    user_id = params["user_id"]
+    type = params["type"]
+    group = Groups.get_group_by_group_id_and_type(params["group_id"], type)
+    new_group_id = group.child_id
+
+    # Fetch the GroupUser with the specified user_id and where the group type is the provided type
+    group_user =
+      GroupUsers.get_group_user_by_user_id_and_type(user_id, type)
+      |> List.first()
+
+    # Fetch the EnrollmentRecord with the specified user_id and where the group_type matches the provided type
+    enrollment_record =
+      from(er in EnrollmentRecord,
+        where: er.user_id == ^user_id and er.group_type == ^type
+      )
+      |> Repo.one()
+
+    # Update both the GroupUser and the EnrollmentRecord
+    with {:ok, %GroupUser{} = updated_group_user} <-
+           GroupUsers.update_group_user(group_user, params),
+         {:ok, %EnrollmentRecord{} = _updated_enrollment_record} <-
+           EnrollmentRecords.update_enrollment_record(enrollment_record, %{
```

Comment:
Shouldn't you check if `enrollment_record` is not `null` or something?

Reply:
Added!